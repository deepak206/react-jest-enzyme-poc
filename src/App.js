import React, { Component, Fragment } from 'react';
import './App.css';
import Header from './components/Header';
import { Link } from 'react-router';

class App extends Component {

  render() {
    return (
      <Fragment><Link to="http://www.instagram.com">Instagram</Link><Header/> </Fragment>
           
    )
  }
}

export default App;