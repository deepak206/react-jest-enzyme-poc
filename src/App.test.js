import React from 'react';
import App from './App';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');

    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const afterClickText = wrapper.find('p').text();
    expect(afterClickText).toEqual('Count: 1');

    const decrementBtn = wrapper.find('button.decrement');
    decrementBtn.simulate('click');
    const decrementText = wrapper.find('p').text();
    expect(decrementText).toEqual('Count: 0');
  });

  //snapshot testing
  it('Snapshot testing', () => {
    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });
});