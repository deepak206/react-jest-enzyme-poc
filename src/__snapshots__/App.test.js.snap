// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<Fragment>
  <Component
    page="http://www.instagram.com"
  >
    Instagram
  </Component>
  <Header />
   
</Fragment>
`;
